tst/EnergyPlus/unit/ZoneContaminantPredictorCorrector.unit.cc

```diff
@@ -390,3 +390,220 @@ TEST_F(EnergyPlusFixture, ZoneContaminantPredictorCorrector_CorrectZoneContamina
     DataContaminantBalance::Contaminant.CO2Simulation = false;
     DataContaminantBalance::Contaminant.GenericContamSimulation = false;
 }
+
+TEST_F(EnergyPlusFixture, ZoneContaminantPredictorCorrector_MultiZoneCO2ControlTest)
+{
+
+    ShortenTimeStepSys = false;
+    UseZoneTimeStepHistory = false;
+
+    ZoneAirHumRat.allocate(3);
+    ZT.allocate(3);
+    MixingMassFlowZone.allocate(3);
+
+    DataGlobals::NumOfZones = 3;
+    DataContaminantBalance::Contaminant.CO2Simulation = true;
+ 
+    DataContaminantBalance::AZ.allocate(3);
+    DataContaminantBalance::BZ.allocate(3);
+    DataContaminantBalance::CZ.allocate(3);
+
+    DataContaminantBalance::CO2ZoneTimeMinus1Temp.allocate(3);
```

Comment:
It would be very nice if these were just member variables of one struct, and you only had to allocate it once, instead of a bunch of dangling arrays that all have to magically be allocated to the same amount... nothing for here, just ranting :laughing: